ui/server/src/main/scala/pl/touk/nussknacker/ui/NussknackerApp.scala

```diff
@@ -71,6 +77,12 @@ trait NusskanckerDefaultAppRouter extends NusskanckerAppRouter {
     val featureTogglesConfig = FeatureTogglesConfig.create(config)
     logger.info(s"Ui config loaded: \nfeatureTogglesConfig: $featureTogglesConfig")
 
+    // TODO: this ugly hack is because we have cycle in dependencies: deploymentService -> repostories -> modelData -> typeToConfig -> deploymentService
+    var deploymentService: DeploymentService = null
+    implicit val getDeploymentService: () => DeploymentService = () => {
```

Comment:
does it have to be implicit? I think it makes it even more confusing :)

Reply:
There are ppl in the Scala World that like using implicit parameters for DI in scala ;-)  but in this place it probable not the best choice - will fix it

Reply:
done